Packs/DuoAdminApi/Integrations/DuoEventCollector/DuoEventCollector.py

```diff
@@ -0,0 +1,217 @@
+from collections import deque
+from enum import Enum
+import duo_client
+from pydantic import BaseModel
+from CommonServerPython import *
+
+
+class LogType(str, Enum):
+    """
+    A list that represent the types of log collecting
+    """
+    AUTHENTICATION = 'AUTHENTICATION'
+    ADMINISTRATION = 'ADMINISTRATION'
+    TELEPHONY = 'TELEPHONY'
+    OFFLINE_ENROLLMENT = 'OFFLINE_ENROLLMENT'
+
+
+class Params(BaseModel):
+    """
+    A class that stores the request params
+    """
+    mintime: dict
+    limit: str = '1000'
+    retries: Optional[str] = '5'
+
+    def set_mintime_value(self, mintime: list, log_type: LogType) -> None:  # pragma: no cover
+        self.mintime[log_type] = mintime
+
+
+class Client:
+    """
+    A class for the client request handling
+    """
+
+    def __init__(self, params: Params):  # pragma: no cover type: ignore
+        self.params = params.get('params')
+        self.admin_api = create_api_call(params.get('host'), params.get('integration_key'),
+                                         (params.get('secret_key')).get('password'))
+
+    def call(self, request_order: list) -> dict:  # pragma: no cover
+        retries = int(self.params.retries)
+        while retries != 0:
+            try:
+                if request_order[0] == LogType.AUTHENTICATION:
+                    response = self.admin_api.get_authentication_log(
+                        mintime=self.params.mintime[LogType.AUTHENTICATION])
+                elif request_order[0] == LogType.ADMINISTRATION:
+                    response = self.admin_api.get_administrator_log(
+                        mintime=self.params.mintime[LogType.ADMINISTRATION])
+                elif request_order[0] == LogType.TELEPHONY:
+                    response = self.admin_api.get_telephony_log(
+                        mintime=self.params.mintime[LogType.TELEPHONY])
+                return response
+            except Exception as exc:
+                msg = f'something went wrong with the sdk call {exc}'
+                LOG(msg)
+                if str(exc) == 'Received 429 Too Many Requests':
+                    retries -= 1
+        return {}
+
+    def set_next_run_filter(self, mintime: int, log_type: LogType):  # pragma: no cover
+        self.params.set_mintime_value(mintime, log_type)
+
+
+class GetEvents:
+    """
+    A class to handle the flow of the integration
+    """
+
+    def __init__(self, client: Client, request_order=[]) -> None:  # pragma: no cover
+        self.client = client
+        self.request_order = request_order
+
+    def rotate_request_order(self) -> None:
+        temp = deque(self.request_order)
+        temp.rotate(-1)
+        self.request_order = list(temp)
```

Comment:
no need for a function
```suggestion
    def rotate_request_order(self) -> None:
        self.request_order = self.request_order[::-1]
```